.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,144 @@
+            }' pom.xml
+        )
+        echo "killbill-oss-parent version=${PARENT_POM_VERSION}"
+        if [[ "$PARENT_POM_VERSION" =~ .*"-SNAPSHOT".* ]] && [ ! -z "$REMOTE_SHA" ]; then
+          echo "::set-output name=FETCH_SNAPSHOT::true"
+        else
+          echo "::set-output name=FETCH_SNAPSHOT::false"
+        fi
+    - name: Checkout killbill-oss-parent
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      uses: actions/checkout@v2
+      with:
+        repository: killbill/killbill-oss-parent
+        ref: ${{ github.ref }}
+        path: killbill-oss-parent
+    - name: Build killbill-oss-parent
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      run: |
+        cd $GITHUB_WORKSPACE/killbill-oss-parent
+        mvn ${MAVEN_FLAGS} clean install -DskipTests=true
+    - name: Checkout killbill-commons
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      uses: actions/checkout@v2
+      with:
+        repository: killbill/killbill-commons
+        ref: ${{ github.ref }}
+        path: killbill-commons
+    - name: Build killbill-commons
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      run: |
+        cd $GITHUB_WORKSPACE/killbill-commons
+        mvn ${MAVEN_FLAGS} clean install -DskipTests=true
+    - name: Checkout killbill-plugin-framework-java
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      uses: actions/checkout@v2
+      with:
+        repository: killbill/killbill-plugin-framework-java
+        ref: ${{ github.ref }}
+        path: killbill-plugin-framework-java
+    - name: Build killbill-plugin-framework-java
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      run: |
+        cd $GITHUB_WORKSPACE/killbill-plugin-framework-java
+        mvn ${MAVEN_FLAGS} clean install -DskipTests=true
+    - name: Checkout killbill-platform
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      uses: actions/checkout@v2
+      with:
+        repository: killbill/killbill-platform
+        ref: ${{ github.ref }}
+        path: killbill-platform
+    - name: Build killbill-platform
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      run: |
+        cd $GITHUB_WORKSPACE/killbill-platform
+        mvn ${MAVEN_FLAGS} clean install -DskipTests=true
+    - name: Checkout killbill-client-java
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      uses: actions/checkout@v2
+      with:
+        repository: killbill/killbill-client-java
+        ref: ${{ github.ref }}
+        path: killbill-client-java
+    - name: Build killbill-client-java
+      if: steps.killbill-oss-parent.outputs.FETCH_SNAPSHOT == 'true'
+      run: |
+        cd $GITHUB_WORKSPACE/killbill-client-java
+        mvn ${MAVEN_FLAGS} clean install -DskipTests=true
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
```

Comment:
Is this step where we add the code into the `codeQL` database -- or does this happen automatically when we run the build?

Reply:
The checkout happens here: https://github.com/killbill/killbill/pull/1462/files/1997e0a7c0f3ec497c1f15e4e023fd0eb0774556#diff-63bd641104d10e25f141d518a16b22a151d125e12701df2f9e79734b23b90188R20